state-and-event-handling/updating-state.md

```diff
@@ -0,0 +1,442 @@
+| `Student`                                      | Presents information about a single student: their `name` and `email`.                                                                    |
+
+<br/>
+
+<details>
+
+<summary>Consider this initial implementation of the <code>Student</code> component.</summary>
+
+```js
+import PropTypes from "prop-types";
+
+const Student = (props) => {
+  return (
+    <li>
+      <ul>
+        <li>Nickname: {props.name}</li>
+        <li>Email: {props.email}</li>
+      </ul>
+    </li>
+  );
+};
+
+Student.propTypes = {
+  name: PropTypes.string.isRequired,
+  email: PropTypes.string.isRequired,
+};
+
+export default Student;
+```
+
+</details>
+
+Sofia wants to add a feature of toggling the presence of a student.
+
+When a student is present, their name is green. When a student is absent, their name is red.
+
+For each student, there is a button to toggle if the student is present or not.
+
+![A web app that lists three students. One student's name is in red. Two students' names are in green.](../assets/state-and-event-handling_updating-state_some-green-1.png)  
+_Fig. Sofia's ideal end result. Ada has a red name because they are absent. Soo-ah and Chrissy have green names because they are present._
+
+She'll need to update her app in the following ways:
+
+1. Create two CSS classes: one that sets the name to green, and another to set the name to red
+1. Create a piece of state in `Student` to hold if the student is present
+1. Create a button in `Student` to toggle if the student is present
+1. Create an event handler that updates state whenever the button is clicked
+1. Use the CSS class in `Student`. One CSS class should be applied if the student is present, and the other CSS class should be applied if they're absent.
+
+### CSS Classes
+
+In `src/components/Student.css`, Sofia adds these two classes:
+
+```css
+.green {
+  background-color: greenyellow;
+}
+
+.red {
+  background-color: lightcoral;
+}
+```
+
+### Create `isPresent` State in `Student`
+
+In `src/components/Student.js`, Sofia adds these two lines to create `isPresent` state:
+
+<!-- prettier-ignore-start -->
+```js
+import { useState } from 'react';
+```
+<!-- prettier-ignore-end -->
+
+This line is in the `Student` component function, before the return statement:
+
+<!-- prettier-ignore-start -->
+```js
+    const [isPresent, setIsPresent] = useState(false);
+```
+<!-- prettier-ignore-end -->
+
+### Create the Toggle Presence Button in `Student`
+
+Sofia updates the returned JSX in `Student`. She adds this `<button>`, with an appropriate label.
+
+<!-- Simon note: I added the indentation to clue to the student that it is likely indented? -->
+<!-- prettier-ignore-start -->
+```js
+    <button>Toggle if {props.name} is present</button>
+```
+<!-- prettier-ignore-end -->
+
```

Comment:
add callout about the need for a wrapping tag
markdown in following comment
no suggestion because code

Reply:
### !callout-danger

## Adjacent JSX elements must be wrapped in an enclosing tag

React tries to update the running web app as Sofia saves her changes, but after saving the `<button>`, React displays the error, "Adjacent JSX elements must be wrapped in an enclosing tag." Sofia realizes that she has violated the rule that React components must return a single parent JSX element!

<br/>

She knows that if she really didn't want to modify her JSX structure further, she could make use of React [Fragments](https://reactjs.org/docs/fragments.html) to group her elements, but instead, she decides to fix this situation by introducing an additional tag.

### !end-callout

To deal with the enclosing tag requirement, Sofia decides to wrap the `Student` JSX in a `<div>` tag, updating the `return` as follows:

```js
    return (
        <div>
            <ul>
                <li>Nickname: {props.name}</li>
                <li>Email: {props.email}</li>
            </ul>
            <button>Toggle if {props.name} is present</button>
        </div>
    );
```

Reply:
Thank you sooooo much for writing this 😭

Reply:
I'll add this commit after merging your branch into this :) I hope this comment doesn't disappear!!